hardware/serial.js

```diff
@@ -2,29 +2,62 @@ var EventEmitter = require('events').EventEmitter;
 var serialport = require('serialport');
 var SerialPort = serialport.SerialPort;
 
+const MAX_RECONNECT_RETRIES = 15;
+const TIME_LIMIT_IN_MILLISECONDS = 60 * 1000;
+
 module.exports = function (device) {
   var serialPort = new SerialPort(device, {
     parser: serialport.parsers.readline("\n"),
     baudrate: 9600
   }, false);
 
-  var emitter = new EventEmitter();
+  var emitter = new EventEmitter(), isConnected = false;
 
-  serialPort.open(function (error) {
-    if (error) {
-        emitter.emit('disconnect');
-    }
-    else {
-      emitter.emit('connected');
-      serialPort.on('data', function (data) {
-        emitter.emit('data', parseInt(data, 10));
-      });
-
-      serialPort.on('error', function () {
-        emitter.emit('disconnect');
-      });
-    }
+  serialPort.on('data', function (data) {
+    emitter.emit('data', parseInt(data, 10));
+  });
+
+  serialPort.on('close', function (data) {
+    isConnected = false;
+    emitter.emit('disconnect');
+    tryReconnectLoop();
+  });
+
+  serialPort.on('open', function (data) {
+    isConnected = true;
```

Comment:
No, it's triggered [on this line](https://github.com/bekk/bekkboard/pull/19/files#diff-05bc479a5d94222306d4b9d7c72c2fdeR46)